src/frontend/src/datafusion/convert.rs

```diff
@@ -90,24 +94,85 @@ pub trait ColumnTrait {
     }
     fn len(&self) -> usize;
     fn column(&self, index: usize) -> Column;
+    fn rw_data_type(&self, index: usize) -> RwDataType;
+    fn df_data_type(&self, index: usize) -> DFDataType;
 }
 
-impl ColumnTrait for DFSchema {
+impl<'a> ColumnTrait for (&'a DFSchema, &'a RwSchema) {
```

Comment:
How about defining a struct instead of using unnamed tuple? It is easier to identify the usage.
```
pub struct InputColumns<'a> {
    df_schema: &'a DFSchema,
    rw_schema: &'a RwSchema,
}
```